lib/base-compiler.ts

```diff
@@ -2756,6 +2930,102 @@ but nothing was dumped. Possible causes are:
         this.supportedLibraries = this.getSupportedLibraries(this.compiler.libsArr, clientOptions.libs[this.lang.id]);
     }
 
+    async getTargetsAsOverrideValues(): Promise<CompilerOverrideOption[]> {
+        if (!this.buildenvsetup || !this.buildenvsetup.getCompilerArch()) {
+            const parser = this.getArgumentParser();
+            const targets = await parser.getPossibleTargets(this);
+
+            return targets.map(target => {
+                return {
+                    name: target,
+                    value: target,
+                };
+            });
+        } else {
+            return [];
+        }
+    }
+
+    async getPossibleStdversAsOverrideValues(): Promise<CompilerOverrideOption[]> {
+        const parser = this.getArgumentParser();
+        return await parser.getPossibleStdvers(this);
+    }
+
+    async populatePossibleOverrides() {
+        if (this.compiler.supportsMarch) {
```

Comment:
It seems like this could be done with less copy paste? As they're mostly identical, perhaps rephrasing as:

```ts
getMaybeArch() {
  if (this.compiler.supportsMarch) return ['-march=vaule']
  if ... etc etc 
  return undefiend; // else
}
//...
const maybeArchArg = this.getMaybeArch();
if (maybeArchArg) {
  const targets = ... the body of one of these;
              if (targets.length > 0) {
                this.compiler.possibleOverrides?.push({
                    name: CompilerOverrideType.arch,
                    display_title: 'Target architecture',
                    description: c_default_target_description,
                    flags: maybeArchStuff,
                    values: targets,
                });
            }
```

then we don't have nearly 100 identical lines for the three cases (and more if we ever have to add more...)